pkg/operator/configobserver/featuregates/observe_featuregates.go

```diff
@@ -70,13 +63,17 @@ func (f *featureFlags) ObserveFeatureFlags(genericListers configobserver.Listers
 		}
 	} else if err != nil {
 		errs = append(errs, err)
-		return prevObservedConfig, errs
+		return existingConfig, errs
 	}
 
 	newConfigValue, err := f.getWhitelistedFeatureNames(configResource)
 	if err != nil {
 		errs = append(errs, err)
-		return prevObservedConfig, errs
+		return existingConfig, errs
+	}
+	currentConfigValue, _, err := unstructured.NestedStringSlice(existingConfig, f.configPath...)
+	if err != nil {
+		errs = append(errs, err)
```

Comment:
shouldn't we `return` on `err`?

Reply:
shouldn't we return `existingConfig` in line `85` in this file? (when we failed to `SetNestedStringSlice`)

Reply:
this case means that the previously observed value was wrongly typed. We have to keep going here to fix that, i.e. not return.

Reply:
ic, it make sense.